packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/index.tsx

```diff
@@ -245,12 +244,17 @@ export default function HierarchyExplorer({ appId, className }: HierarchyExplore
 
       const fragment = appDom.cloneFragment(dom, nodeId);
 
-      const updatedDom = appDom.addFragment(dom, fragment, node.parentId, node.parentProp);
-
       const newNode = appDom.getNode(fragment, fragment.root);
       const editorDomView = getNodeEditorDomView(newNode);
 
-      domApi.update(updatedDom, editorDomView || { kind: 'page' });
+      domApi.update((draft) => {
+        invariant(
```

Comment:
Is this still necessary if we wrap the whole function in `domAPi.update`?

Reply:
This invariant was outside `domApi.update` before, but if it's not inside the function it doesn't stop the type checker from complaining.
I just tried to write this function differently, but because I've made the second argument of `domApi.update` be the view we want to change to, that view needs to be set before calling `domApi.update`, so it needs to be outside of it, along with everything else that value depends on.